src/App.js

```diff
@@ -1,39 +1,190 @@
-import './App.css';
+import "./scss/styles.scss"
+import { Routes, Route } from "react-router-dom"
+import NavBar from "./components/NavBar"
+import Product from "./components/Product"
+import Header from "./components/Header"
+import Footer from "./components/Footer"
+import ProductDetails from "./components/ProductDetail"
+import HeaderNav from "./components/HeaderNav"
+import axios from "axios"
+
+import { useState, useEffect, useRef } from "react"
 
 function App() {
+  const [product, setProduct] = useState("placeholder")
+  const [filteredArray, setFilteredArray] = useState([])
+  const [arrayOfObjectCart, setArrayOfObjectCart] = useState([])
+  const [isCartClicked, setIsCartClicked] = useState(false)
+  const [isHeartClicked, setIsHeartClicked] = useState(false)
+  const [arrayOfObjectWish, setArrayOfObjectWish] = useState([])
+  const scollToRef = useRef();
+  const [wishList, setWishList] = useState([])
 
-  async function getApiData(){
-    const response = await fetch(
-      `http://makeup-api.herokuapp.com/api/v1/products.json`
-    )
-    const data = await response.json()
-    console.log(data)
-    return data
-  }
-  getApiData()
-  return (
-    <div className="App">
+  useEffect(() => {
+    const myparams = product==="placeholder"?
+    {}:
+    {
+      product_type: `${product}`
+    }
+    axios
+      .get("https://makeup-api.herokuapp.com/api/v1/products.json", {
+        params: myparams 
+      })
+      .then(function (res) {
+        const filteredProduct1 = res.data
+          .filter(
+            (individual) =>
+              individual.currency === "USD" &&
+              individual.price !== "0.0" &&
+              individual.id !== 1005 &&
+              individual.id !== 1004 &&
+              individual.id !== 1003 &&
+              individual.id !== 1002 &&
+              individual.id !== 1001 &&
+              individual.id !== 1000 &&
+              individual.id !== 999
+          )
+          .slice(0, 200)
+          filteredProduct1.map((item) => {
+            item.inWishList = false
+            arrayOfObjectWish.filter((x)=> {
```

Comment:
To understand the code more easily, try changing "x" to another variable with meaning :)